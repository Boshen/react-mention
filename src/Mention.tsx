import * as React from 'react'

export class Mention extends React.Component<{}, {}> {

  render() {
    return (
      <div>React Mention</div>
    )
  }

}
