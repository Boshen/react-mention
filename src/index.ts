export { Mention, CaretPosition } from './Mention'
export { Trigger } from './Trigger'
