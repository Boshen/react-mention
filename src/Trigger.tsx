import * as React from 'react'
import * as PropTypes from 'prop-types'

import { CaretPosition } from './Mention'

export type TriggerProps = {
  value: string
  action: (pos?: CaretPosition) => void
}

export class Trigger extends React.PureComponent<TriggerProps, {}> {

  static propTypes = {
    value: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired
  }

  render() {
    return (
      <div></div>
    )
  }

}
