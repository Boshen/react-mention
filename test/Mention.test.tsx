import * as React from 'react'
const renderer = require('react-test-renderer')

import { Mention } from '../src/Mention'

describe('Mention Snapshot', () => {

  it('renders correctly', () => {
    const component = (
      <Mention />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
