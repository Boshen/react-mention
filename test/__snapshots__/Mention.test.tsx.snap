// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention Snapshot renders correctly 1`] = `
<div>
  React Mention
</div>
`;
